vcd/resource_vcd_provider_vdc.go

```diff
@@ -0,0 +1,688 @@
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed:    true,
+				Description: "Set of IDs to the storage profiles available to this Provider VDC",
+			},
+			"resource_pool_ids": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Required:    true,
+				Description: "Set of IDs of the resource pools backing this provider VDC",
+			},
+			"network_pool_ids": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Computed:    true,
+				Description: "Set IDs of the network pools used by this Provider VDC",
+			},
+			"universal_network_pool_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the universal network reference",
+			},
+			"host_ids": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Computed:    true,
+				Description: "Set with all the hosts which are connected to VC server",
+			},
+			"vcenter_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the vCenter server that provides the resource pools and datastores",
+			},
+			// TODO: metadata handling to be added after refactoring of conflicting fields "metadata" and "metadata_entry"
+			//"metadata": {
+			//	Type:        schema.TypeMap,
+			//	Computed:    true,
+			//	Description: "Key and value pairs for Provider VDC metadata",
+			//	Deprecated:  "Use metadata_entry instead",
+			//},
+			//"metadata_entry": getMetadataEntrySchema("Provider VDC", false),
+		},
+	}
+}
+
+func resourceVcdProviderVdcCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	providerVdcName := d.Get("name").(string)
+	providerVdcDescription := d.Get("description").(string)
+	vcenterId := d.Get("vcenter_id").(string)
+	managerid := d.Get("nsxt_manager_id").(string)
+	hwVersion := d.Get("highest_supported_hardware_version").(string)
+	rawNetworkPoolIds := d.Get("network_pool_ids").(*schema.Set)
+	rawResourcePoolIds := d.Get("resource_pool_ids").(*schema.Set)
+	managerHref, err := url.JoinPath(vcdClient.Client.VCDHREF.String(), "admin", "extension", "nsxtManagers", managerid)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	vcenter, err := vcdClient.GetVcenterById(vcenterId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	vcenterUrl, err := vcenter.GetVimServerUrl()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	if rawNetworkPoolIds.Len() == 0 {
+		return diag.Errorf("no network pool was provided")
+	}
+	if rawNetworkPoolIds.Len() > 1 {
+		return diag.Errorf("only one network pool can be used to create a Provider VDC")
+	}
+	if rawResourcePoolIds.Len() == 0 {
+		return diag.Errorf("no resource pool was provided")
+	}
+	if rawResourcePoolIds.Len() > 1 {
+		return diag.Errorf("only one resource pool can be used to create a Provider VDC")
+	}
```

Comment:
Should these validations be put directly in schema?

Reply:
Can't be done: we can only insert one network pool during creation, but the provider VDC can later (and without user intervention) acquire more.